pkg/v3/tickers/recovery.go

```diff
@@ -1,19 +1,28 @@
 package tickers
 
 import (
+	"sync"
 	"time"
 
 	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
 )
 
 type recoveryTicker struct {
 	*retryTicker
-	// lastBlock ocr2keepers.BlockKey
+	lock      sync.RWMutex
+	lastBlock ocr2keepers.BlockKey
 }
 
-// getterFn is a function that retrieves the recoveryTick for the given time.
+// setBlock updates the block of the given payload
+func (rt *recoveryTicker) setBlock(p ocr2keepers.UpkeepPayload) {
```

Comment:
The goal of this function is to be exported on the ticker such that an outside component can set the block. Something like `SetBlock(UpkeepKey)`. It is not intended to be used by the ticker itself.